lib/symboldatabase.cpp

```diff
@@ -4253,20 +4288,23 @@ bool Scope::isVariableDeclaration(const Token* const tok, const Token*& vartok,
     if (Token::Match(localVarTok, "%name% ;|=") || (localVarTok && localVarTok->varId() && localVarTok->strAt(1) == ":")) {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
     } else if (Token::Match(localVarTok, "%name% )|[") && localVarTok->str() != "operator") {
         vartok = localVarTok;
         typetok = localTypeTok;
+        return nullptr != vartok && vartok->isName();
```

Comment:
I think some refactorings would be nice;
 * could we avoid copy paste in some nice way here?
 * Do not assign `vartok = localVarTok` and then return false. If false is returned we should either leave vartok/typetok as they are or assign nullptr.

Reply:
I've refactored this as requested in 1f09ab34da759cfe8e3ca2c948bb82639ae57e71